### PR TITLE
Keep immutability intact across calls to ConvertToVectorRep(NC)

### PR DESCRIPTION
**Pull request by markuspf**

### CODE REVIEW

**Comment by markuspf** [comment]
This will require #1645 to work properly, because I switch to using `SWITCH_OBJ` instead of `CLONE_OBJ`.

File: hpcgap/lib/vecmat.gi [hunk 1]
```diff
@@ -1305,6 +1305,11 @@ InstallGlobalFunction(ConvertToVectorRepNC,function( arg )
                 #
                 return true;
             fi;
+            # Switching the object below can change the mutability of
+            # v, hence restore it
```
Comment (fingolfin) [inline hpcgap/lib/vecmat.gi]: But the object is not actually switched, is it?

File: lib/vecmat.gi [hunk 2]
```diff
@@ -1280,7 +1280,12 @@ InstallGlobalFunction(ConvertToVectorRepNC,function( arg )
                 #
                 return true;
             fi;
-            CLONE_OBJ(v,vc); # horrible hack.
+            # SWITCH_OBJ changes the mutability of v, so we make
+            # sure that if v is immutable it stays immutable.
+            if not IsMutable(v) then
+                MakeImmutable(vc);
+            fi;
+            SWITCH_OBJ(v,vc);
```
Comment (fingolfin) [inline lib/vecmat.gi]: Aha... So, does this not work in HPC-GAP? I notice that there, the code still uses `CLONE_OBJ`

**Comment by markuspf** [comment]
This is rebased and should be merged.

**Comment by fingolfin** [comment]
Let's let the tests complete one last time, then merge

**Pull request merged**